lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1455,6 +1455,17 @@ static LogicalResult verifyPortSymbolUses(FModuleLike module,
   // verify types in ports.
   for (size_t i = 0, e = module.getNumPorts(); i < e; ++i) {
     auto type = module.getPortType(i);
+
+    if (auto refType = dyn_cast<RefType>(type)) {
+      auto layer = refType.getLayer();
+      if (layer && !dyn_cast_or_null<LayerOp>(
+                       symbolTable.lookupSymbolIn(circuitOp, layer)))
```

Comment:
`symbolTable.lookupSymbolIn<LayerOp>` does the `dyn_cast_or_null` for you.

Reply:
Fixed in https://github.com/llvm/circt/pull/6574/commits/68b4e01b15af105b51776e0225616d1cf8853e38.